src/views/Main/Catalog/Components/Schema.js

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react';
+import { Button } from 'react-bootstrap';
+import '../styles.module.css';
+import { PropertyTypeList } from './PropertyTypeList';
+import CatalogApi from '../../../../utils/CatalogApi';
+import Consts from '../../../../utils/AppConsts';
+import { EntityTypeFqnList } from './EntityTypeFqnList';
+
+export class Schema extends React.Component {
+  static propTypes = {
+    name: PropTypes.string,
+    namespace: PropTypes.string,
+    propertyTypes: PropTypes.string,
+    entityTypeFqns: PropTypes.string
+  }
+
+  constructor() {
+    super();
+    this.handleClick = this.handleClick.bind(this);
+    this.downloadFile = this.downloadFile.bind(this);
+    this.displayError = this.displayError.bind(this);
+    this.enableButton = this.enableButton.bind(this);
+    this.state = {
+      error: Consts.ERROR_STATE.hide,
+      disableJson: false
+    };
+  }
+
+  handleClick() {
+    this.downloadFile(Consts.JSON);
+    this.setState({ disableJson: true });
+  }
+
+  downloadFile(datatype) {
+    CatalogApi.downloadSchema(
+      this.props.name,
+      datatype,
+      this.props.entityTypeFqns,
+      this.displayError,
+      this.enableButton
+    );
+  }
+
+  displayError() {
+    this.setState({
+      error: Consts.ERROR_STATE.show,
+      disableJson: false
+    });
+  }
+
+  enableButton() {
+    this.setState({ disableJson: false });
+  }
+
+  render() {
```

Comment:
this `render()` looks very similar to `render()` in `src/views/Main/Catalog/Components/EntityType.js`.

this is fine for now, but I think it's worth trying to extract a common component that is responsible for just the markup, and here, you'll just pass in whatever is needed that is custom to `Schema`.